### PR TITLE
Feature/aa 255 add endpoint for change notification options for current agent

### PR DESCRIPTION
**Pull request by AlexeyShok**
Added endpoint for change notification options for current agent

### CODE REVIEW

**Review by hasmikharutyunyan — APPROVED** [review]

File: Api/Controllers/AgentControllers/NotificationController.cs [hunk 1]
```diff
@@ -58,6 +58,23 @@ public async Task<IActionResult> GetNotificationOptions()
         }
 
 
+        /// <summary>
+        ///     Updates the notification options of the current agent
+        /// </summary>
+        /// <param name="notificationOptions">Notification options</param>
+        /// <returns></returns>
+        [HttpPut("options")]
```
Comment (kirillta) [inline Api/Controllers/AgentControllers/NotificationController.cs:66]: Not sure if one can differentiate `settings` from `options`

File: Api/NotificationCenter/Services/NotificationOptionsService.cs [hunk 2]
```diff
@@ -53,7 +53,28 @@ public async Task<Result<SlimNotificationOptions>> GetNotificationOptions(int us
...
+        public async Task<Result> Update(SlimAdminContext admin, Dictionary<NotificationTypes, SlimNotificationOptions> notificationOptions)
+            => await Update(admin.AdminId, ApiCallerTypes.Admin, null, notificationOptions);
+
+
+        private async Task<Result> Update(int userId, ApiCallerTypes userType, int? agencyId, Dictionary<NotificationTypes, SlimNotificationOptions> notificationOptions)
+        {
+            foreach (var option in notificationOptions)
+            {
+                var result = await Update(userId, userType, agencyId, option.Key, option.Value);
+                if (result.IsFailure)
```
Comment (kirillta) [inline Api/NotificationCenter/Services/NotificationOptionsService.cs:69]: Should we fail after one failure here?

File: Api/Controllers/AgentControllers/NotificationController.cs (hunk 1 above)
Comment (AlexeyShok) [reply Api/Controllers/AgentControllers/NotificationController.cs]: Renamed

File: Api/NotificationCenter/Services/NotificationOptionsService.cs (hunk 2 above)
Comment (AlexeyShok) [reply Api/NotificationCenter/Services/NotificationOptionsService.cs]: Fixed

**Pull request merged**